src/9-v04x-migration/index.md

```diff
@@ -0,0 +1,92 @@
+---
+date: 2016-02-12
+id: 9-v04x-migration
+template: tmpl/layouts/post.html
+baseurl: ..
+breadcrumbs:
+  - {name: "9-v04x-migration", link: "./" }
+tags: gateway, bootstrap, infrastructure
+title: Migrating ipfs.io from go-ipfs 0.3.x to 0.4.0
+author: Lars Gierth
+collection: posts
+---
+
+Good news everyone! We are about to release go-ipfs 0.4.0,
+which contains lots of great changes and enhancements.
+
+There are breaking changes
+which prevent 0.4.x nodes from communicating with 0.3.x nodes.
+This means that there will be, and already are, two separate networks.
+We'll call these networks v03x and v04x.
+IPFS nodes built from the master branch are already part of v04x.
+The Docker image `ipfs/go-ipfs` is built from master and thus also part of v04x.
+
+ipfs.io provides two essential services to the community,
+which are affected by this: the gateway and the default bootstrappers.
+We'll continue to support them for the v03x network until the **end of April 2016**.
+
+Please note that we won't support 0.3.x with patches or new features.
+All development effort is directed towards 0.4.0, and you should update as soon as possible.
+
+- [How do I update go-ipfs to 0.4.0?](#how-do-i-update-go-ipfs-to-0-4-0-)
+- [The public gateways at ipfs.io](#the-public-gateways-at-ipfs-io)
+- [The default bootstrappers](#the-default-bootstrappers)
+- [How does this affect me?](#how-does-this-affect-me-)
+
+
+## How do I update go-ipfs to 0.4.0?
+
+You can use the [ipfs-update tool][ipfs-update] to update go-ipfs.
+
+[ipfs-update]: http://dist.ipfs.io/#ipfs-update
+
+
+## The public gateways at ipfs.io
+
+For the time being, https://ipfs.io uses both v03x and v04x to service requests.
+Whichever responds successfully first (2xx or 3xx status code),
+gets to serve its content. The other response is discarded.
+If the first response is not successful (4xx/5xx, connection errors),
+it is discarded and the second response is served, regardless of its status code.
+
+We're using the [multireq proxy][multireq] to accomplish this multiplexing behaviour.
```

Comment:
Add:
"`multireq` is a new tool, so if you notice any weirdness with gateway requests, [please let us know](https://github.com/ipfs/infrastructure/issues/)."